### ISSUE TITLE
3.0.0 is depending on outdated permission_handler package

### ISSUE THREAD

**Issue by Mickco** [themobilecoder/headset_connection_event#17]
First of all, I really love your work.
Would you consider upgrade permission_handler to a newer version, or update a headset_connection_event without permission_handler package. Thanks a lot.
```
Because headset_connection_event >=3.0.0 depends on permission_handler ^10.0.1 and you_arrived_app depends on permission_handler ^11.0.1, headset_connection_event >=3.0.0 is forbidden.
So, because you_arrived_app depends on headset_connection_event ^3.0.0, version solving failed.


You can try one of the following suggestions to make the pubspec resolve:
* Consider downgrading your constraint on headset_connection_event: flutter pub add headset_connection_event:^2.1.1
* Consider downgrading your constraint on permission_handler: flutter pub add permission_handler:^10.4.5
```

**Comment by rafdls** [comment]
Hi @Mickco,

Please see the #18 pull request if that looks okay. 

When you get the time, you can first try it on your project with:
```
  headset_connection_event:
    git:
      url: git@github.com:themobilecoder/headset_connection_event.git
      ref: upgrade-permission-handler-package
```

I tested this change on Android 13 and permission for Bluetooth works.

Once merged, ill publish this update in pub.dev.

Cheers,
Rafael

**Comment by rafdls** [comment]
Fixed by #18
